src/xref2/compile.ml

```diff
@@ -353,30 +371,35 @@ and functor_parameter : Env.t -> FunctorParameter.t -> FunctorParameter.t =
 and functor_parameter_parameter :
     Env.t -> FunctorParameter.parameter -> FunctorParameter.parameter =
  fun env' a ->
-  let exception Compile_functor_parameter_parameter in
-
-  let functor_arg, env =
-    match Env.lookup_module' a.id env' with
-    | Some (_, _ as x) -> x
-    | None -> raise Compile_functor_parameter_parameter
+  let open Utils.ResultMonad in
+  let get_module_type_expr = function
+    | Component.Module.ModuleType expr -> Ok expr
+    | _ -> Error `Resolve_type
```

Comment:
Resolve_type should probably be `Resolve_module_type` - but again this is an internal error - I don't think it is possible for functors to have opaque argument types.